client_opts.go

```diff
@@ -33,25 +33,66 @@ func WithDialOpts(opts []grpc.DialOption) ClientOpt {
 	}
 }
 
-// RemoteOpts allows the caller to set distribution options for a remote
-type RemoteOpts func(*Client, *RemoteContext) error
+// RemoteOpt allows the caller to set distribution options for a remote
+type RemoteOpt func(*Client, *RemoteContext) error
+
+type pullOptions struct {
+	labels map[string]string
```

Comment:
Could we just add `Labels` to `RemoteContext`? Many of the values on there are already pull specific, I don't think it makes sense to split the options just for the labels without splitting them all. We had a unified push/pull type to keep the options simpler and used the `WithPull*` nomenclature to differentiate. The fact that this necessitates `WithPullRemoteOpt` indicates to me that it should not exist.

Reply:
Sure